src/attachment/index.ts

```diff
@@ -62,7 +76,15 @@ let exportAttachment;
       const validProps = validateProps(props);
       Object.assign(this, validProps);
     }
-
+    shouldUpdate(changedProperties: PropertyValues): boolean {
+      if (changedProperties.has("files")) {
+        if (!isArrayType<FileItem>(this.files)) {
```

Comment:
you can combine the condition:
if (
        changedProperties.has("files") &&
        !isArrayType<FileItem>(this.files)
      ) {
        throwErrorAfterUpdateComplete(this, ERROR_MESSAGE.FILES.IS_NOT_ARRAY);
        return false;
      }